### PR TITLE
NIFI-8630 Upgraded javax.mail 1.4.7 to jakarta.mail 2.0.1 for PutEmail

### PR DESCRIPTION
**Pull request by exceptionfactory**
#### Description of PR

NIFI-8630 Upgrades `javax.mail` 1.4.7 to `jakarta.mail` 2.0.1 to support modern TLS protocol versions in the `PutEmail` processor.

Older versions of javax.mail do not support TLSv1.2, causing `SSLHandshakeException` errors in `PutEmail` when running on Java versions that have disabled TLSv1 and TLSv1.1.  The `jakarta.mail` library is the maintained and repackaged version of the `javax.mail` library, which supports current TLS protocol versions matching JDK enabled protocols.  Upgrading to Jakarta Mail also required including the updated Jarkata Activation library.

Additional updates include a display name of `SMTP STARTTLS` for the `SMTP TLS` property, indicating that it is used to enable Opportunistic TLS over SMTP.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically `main`)?

- [X] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [X] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [X] Have you verified that the full build is successful on JDK 11?
- [X] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [X] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [X] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pvillard31** [comment]
Do we know if this also solves https://issues.apache.org/jira/browse/NIFI-8281 ?
(and maybe https://issues.apache.org/jira/browse/NIFI-6961)

**Comment by exceptionfactory** [comment]
> Do we know if this also solves https://issues.apache.org/jira/browse/NIFI-8281 ?
> (and maybe https://issues.apache.org/jira/browse/NIFI-6961)

Thanks for referencing those issues @pvillard31.  This PR should resolve NIFI-8281 since it includes the new version of Jakarta Activation.  The dependency changes include both the Jakarta Activation API as well as the Jakarta Activation implementation library.

NIFI-6961 references the `ExtractEmailHeaders` processor, which is in a different NAR bundle, so a separate PR will be necessary to address dependencies in that processor.  Upgrading dependencies in that bundle will be more challenging due to transitive dependencies on `javax.mail` from several libraries, but making a similar change is necessary.

I updated both of those Jira issues with links to NIFI-8630.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java [hunk 1]
```diff
@@ -437,19 +431,18 @@ public void process(InputStream in) throws IOException {
      */
     private Session createMailSession(final Properties properties) {
         String authValue = properties.getProperty("mail.smtp.auth");
```
Comment (gresockj) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java:433]: I notice a few other local variables that are not final but could be.  What do you think about making them final in this PR?

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java [hunk 2]
```diff
@@ -437,19 +431,18 @@ public void process(InputStream in) throws IOException {
...
+        final boolean auth = Boolean.parseBoolean(authValue);
 
         /*
          * Conditionally create a password authenticator if the 'auth' parameter is set.
          */
-        final Session mailSession = auth ? Session.getInstance(properties, new Authenticator() {
+        return auth ? Session.getInstance(properties, new Authenticator() {
             @Override
             public PasswordAuthentication getPasswordAuthentication() {
                 String username = properties.getProperty("mail.smtp.user"), password = properties.getProperty("mail.smtp.password");
```
Comment (gresockj) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutEmail.java:442]: Quaint use of the multi-variable declaration in the existing code -- how do you feel about splitting this into two lines?

**Review by gresockj — COMMENTED** [review]
I don't see any substantive issues with the changes, just a few minor style suggestions for the existing code.  Planning to verify runtime behavior next.

**Review by gresockj — APPROVED** [review]
Using ListenSMTP over TLSv1.2 and TLSv1.3, I verified that PutEmail now works with these protocols.  Nice work!

**Review by pvillard31 — APPROVED** [review]

**Pull request closed without merging**